### PR TITLE
[JENKINS-36720] Spotbugs issues

### PR DESCRIPTION
**Pull request by StefanSpieker**
Fixed Spotbugs issues:
* [RV_NEGATING_RESULT_OF_COMPARETO](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-negating-the-result-of-compareto-compare-rv-negating-result-of-compareto)
* [RV_RETURN_VALUE_IGNORED_BAD_PRACTICE](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-method-ignores-exceptional-return-value-rv-return-value-ignored-bad-practice)

See [JENKINS-36720](https://issues.jenkins-ci.org/browse/JENKINS-36720).

### Proposed changelog entries

* Internal: N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/WebAppMain.java
Comment (MarkEWaite) [inline core/src/main/java/hudson/WebAppMain.java:259]: I believe that the WARNING level is on by default.

Wouldn't it be safer and less likely to spam the log if this were logged at a level that is off by default, like FINE?  Since the failure to delete has not been a reported problem in the past, it seems wise to not risk log spam on systems like Windows.
```suggestion
                    LOGGER.log(WARNING, "Temp file test.test could not be deleted.");
```

**Review by MarkEWaite — COMMENTED** [review]
Seems risky on Windows file systems with virus scanners to default to writing the message to the log when failure to delete the file appears harmless to the overall function of that method.

File: core/src/main/java/hudson/WebAppMain.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/WebAppMain.java]: Thanks Mark, I changed it as suggested 👍

**Review by res0nance — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]

File: core/src/main/java/hudson/WebAppMain.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/WebAppMain.java:259]: Should it throw exception here? I read the comment, and it looks like it would be safer to report the error

**Review by alecharp — APPROVED** [review]
:+1:

**Review by varyvol — APPROVED** [review]

**Comment by alecharp** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
